scripts/create_newcase

```diff
@@ -1,1003 +1,177 @@
-    my $comps_wav  = $config->get("COMP_WAV");
-    my @comps  = ($comps_atm, $comps_lnd, $comps_ocn, $comps_ice, $comps_glc, $comps_rof, $comps_wav);
-
-    # Create sourcemods
-    foreach my $comp (@comps)  {
-	mkpath("$caseroot/SourceMods/src.$comp");
-
-	# Note: this is CESM specific given that we are referencing cism explitly
-	if ($comp eq 'cism') {
-	    mkpath("$caseroot/SourceMods/src.cism/glimmer-cism");
-	    open(README, ">", "$caseroot/SourceMods/src.$comp/README") or
-		$logger->logdie( "cannot open > $caseroot/SourceMods/src.cism/README: $!");
-	    print README
-		"Put source mods for the glimmer-cism library in the glimmer-cism subdirectory.\n",
-		"This includes any files that are in the glimmer-cism subdirectory of $cimeroot/../components/cism.\n\n",
-		"Anything else (e.g., mods to source_glc or drivers) goes in this directory, NOT in glimmer-cism/.\n";
-	    close(README);
-	}
-
-    }
-    mkpath("$caseroot/SourceMods/src.share");
-    mkpath("$caseroot/SourceMods/src.drv");
-}
-
-#-------------------------------------------------------------------------------
-sub _set_test
-{
-    # Set the parameters for the specified testname.  The
-    # parameters are read from an input file, and if no testname matches are
-    # found then issue error message.
-    # This routine uses the configuration defined at the package level ($config).
-
-    my ($test_file, $testname, $config) = @_;
-
-    my $parser = XML::LibXML->new( no_blanks => 1);
-    my $xml = $parser->parse_file($test_file);
-    my @nodes = $xml->findnodes(".//test[\@NAME=\"$testname\"]/*");
-    if (@nodes) {
-	foreach my $node (@nodes) {
-	    my $id = $node->nodeName();
-	    my $value = $node->textContent();
-	    $config->set($id, $value);
-	}
-    } else {
-	$logger->warn("_set_test: no match for test $testname - possible testnames are:");
-	@nodes = $xml->findnodes(".//test");
-	foreach my $node (@nodes) {
-	    my $name = $node->getAttribute('NAME');
-	    my @child_nodes = $node->findnodes('./DESC');
-	    my $desc = $child_nodes[0]->textContent();
-	    $logger->warn( " $name ($desc) ") ;
-	}
-	$logger->logdie( "_set_test: exiting");
-    }
-}
-
-#-------------------------------------------------------------------------------
-sub _clean
-{
-    my ($name) = @_;
-
-    $name =~ s/^\s+//; # strip any leading whitespace
-    $name =~ s/\s+$//; # strip any trailing whitespace
-    return ($name);
-}
+#!/usr/bin/env python2
 
+from Tools.standard_script_setup import *
 
+from CIME.utils         import expect, get_model
+from CIME.case          import Case
+
+import stat
+
+logger = logging.getLogger(__name__)
+
+###############################################################################
+def parse_command_line(args):
+###############################################################################
+
+    cime_model = CIME.utils.get_model()
+
+    parser = argparse.ArgumentParser()
+
+    CIME.utils.setup_standard_logging_options(parser)
+
```

Comment:
Can we get rid of the non compliant "-longoption"? I know this may require that you chase down all the old calls to create_newcase, but I don't think they should be too many of those. We could then use proper short options which helps usability.